pyccel/parser/semantic.py

```diff
@@ -492,11 +499,21 @@ def remove_variable(self, name):
     def get_header(self, name):
         """."""
         container = self.namespace
+        headers = []
         while container:
             if name in container.headers:
-                return container.headers[name]
+                headers += container.headers[name]
             container = container.parent_scope
-        return None
+        return headers
+
+    def get_templates(self):
+        """."""
+        container = self.namespace
+        templates = []
+        while container:
+            templates += container.templates
+            container = container.parent_scope
+        return templates
```

Comment:
Does this handle name collisions in different scopes?
```
@template('T',['int','real','complex'])
...
def bla():
    @template('T',['int','real'])
    ...
```

Reply:
now it's handled, a template overrides any templates with the same name in its parents scopes, although we can't test it since neither classes nor nested functions are implemented yet

Reply:
nested functions are implemented in fortran

Reply:
ok perfect I will add tests in fortran